apps/web/src/pages/templates/workflow/workflow/node-types/ChannelNode.tsx

```diff
@@ -15,31 +17,30 @@ interface NodeData {
   active?: boolean;
 }
 
-export default memo(
-  ({ data, selected, id, dragging }: { data: NodeData; selected: boolean; id: string; dragging: boolean }) => {
-    const { getNode, getEdges, getNodes } = useReactFlow();
-    const thisNode = getNode(id);
-    const isParent = thisNode ? getOutgoers(thisNode, getNodes(), getEdges()).length : false;
-    const noChildStyle = isParent ? {} : { border: 'none', background: 'transparent' };
+export default memo(({ data, id, dragging }: { data: NodeData; selected: boolean; id: string; dragging: boolean }) => {
+  const { getNode, getEdges, getNodes } = useReactFlow();
+  const { selectedNodeId } = useTemplateEditorContext();
+  const thisNode = getNode(id);
+  const isParent = thisNode ? getOutgoers(thisNode, getNodes(), getEdges()).length : false;
+  const noChildStyle = isParent ? {} : { border: 'none', background: 'transparent' };
 
-    return (
-      <div data-test-id={`node-${data.testId}`} style={{ pointerEvents: 'none' }}>
-        <ChannelButton
-          setActivePage={data.setActivePage}
-          errors={data.error}
-          onDelete={data.onDelete}
-          tabKey={data.tabKey}
-          Icon={data.Icon}
-          label={data.label}
-          active={selected}
-          disabled={!data.active}
-          id={id}
-          index={data.index}
-          dragging={dragging}
-        />
-        <Handle type="target" id="b" position={Position.Top} />
-        <Handle style={noChildStyle} type="source" id="a" position={Position.Bottom} />
-      </div>
-    );
-  }
-);
+  return (
+    <div data-test-id={`node-${data.testId}`} style={{ pointerEvents: 'none' }}>
+      <ChannelButton
+        setActivePage={data.setActivePage}
+        errors={data.error}
+        onDelete={data.onDelete}
+        tabKey={data.tabKey}
+        Icon={data.Icon}
+        label={data.label}
+        active={id === selectedNodeId}
```

Comment:
What only changed here is this like, we don't use the internally selected state of the library, but our state field `selectedNodeId` for highlighting the node. 

This code allows us to select the active node in the beginning when the workflow tree editor is loaded, without it we are not able to do that.